components/do_verify_email/do_verify_email.tsx

```diff
@@ -55,7 +55,16 @@ export default class DoVerifyEmail extends React.PureComponent<Props, State> {
         if (this.props.isLoggedIn) {
             GlobalActions.redirectUserToDefaultTeam();
         } else {
-            browserHistory.push('/login?extra=verified&email=' + encodeURIComponent((new URLSearchParams(this.props.location.search)).get('email') || ''));
+            let link = '/login?extra=verified';
+            const email = encodeURIComponent((new URLSearchParams(this.props.location.search)).get('email') || '');
+            if (email) {
+                link += '&email=' + email;
```

Comment:
I think we can do the `encodeURIComponent` inside the if block to avoid the use of  `||`.